blosc2/blosc2_ext.pyx

```diff
@@ -837,6 +839,52 @@ cdef class SChunk:
             raise RuntimeError("Could not update the desired chunk")
         return rc
 
+    def get_slice(self, start=0, stop=None, out=None):
+        start, stop = self._massage_key(start, stop)
+        cdef int nbytes = (stop - start) * self.schunk.typesize
+        cdef Py_buffer *buf
+        if out is not None:
+            buf = <Py_buffer *> malloc(sizeof(Py_buffer))
+            PyObject_GetBuffer(out, buf, PyBUF_SIMPLE)
+            if buf.len < nbytes:
+                raise ValueError("Not enough space for writing the slice in out")
+            rc = blosc2_schunk_get_slice_buffer(self.schunk, start, stop, <void *> buf.buf)
+            PyBuffer_Release(buf)
+        else:
+            out = PyBytes_FromStringAndSize(NULL, nbytes)
+            if out is None:
+                raise RuntimeError("Could not get a bytes object")
+            rc = blosc2_schunk_get_slice_buffer(self.schunk, start, stop, <void*><char *> out)
+            if rc >= 0:
+                return out
+        if rc < 0:
+            raise RuntimeError("Error while getting the slice")
+
+    def set_slice(self, value, start=0, stop=None):
+        start, stop = self._massage_key(start, stop)
+        cdef int nbytes = (stop - start) * self.schunk.typesize
+        cdef Py_buffer *buf = <Py_buffer *> malloc(sizeof(Py_buffer))
+        PyObject_GetBuffer(value, buf, PyBUF_SIMPLE)
+        if buf.len < nbytes:
+            raise ValueError("Not enough data for writing the slice")
+        rc = blosc2_schunk_set_slice_buffer(self.schunk, start, stop, <void *> buf.buf)
+        PyBuffer_Release(buf)
+        if rc < 0:
+            raise RuntimeError("Error while setting the slice")
+
+    def _massage_key(self, start, stop):
+        if stop is None:
+            stop = self.schunk.nbytes / self.schunk.typesize
```

Comment:
`/` gives a float as a result.  Better use `//`.

Reply:
Fixed in 20b3d298e5aef2b2731aabf104d51dc0a67dcaca.